modules/terraform-zscc-asg-lambda-aws/main.tf

```diff
@@ -0,0 +1,284 @@
+################################################################################
+#  Pull the Account ID number of the account that owns or contains the calling entity.
+################################################################################
+data "aws_caller_identity" "asg_caller_identity" {}
+
+################################################################################
+# Create IAM Role and Policy for Lambda
+################################################################################
+resource "aws_iam_role" "asg_lambda_iam_role" {
+  name = "${var.name_prefix}-asg-lambda-iam-role-${var.resource_tag}"
+
+  assume_role_policy = <<EOF
+{
+  "Version": "2012-10-17",
+  "Statement": [
+    {
+      "Action": "sts:AssumeRole",
+      "Principal": {
+        "Service": "lambda.amazonaws.com"
+      },
+      "Effect": "Allow",
+      "Sid": ""
+    }
+  ]
+}
+EOF
+}
+
+
+################################################################################
+# Define AWS Managed Secrets Manager Get Secrets Policy
+################################################################################
+# Retrieve Secret Manager ARN by friendly name
+data "aws_secretsmanager_secret" "lambda_secret_name" {
```

Comment:
Are you basically creating copies of the Cc IAM Policies and not re-using them for lambda?